proposals/readonly-methods.md

```diff
@@ -0,0 +1,65 @@
+# Readonly Methods
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Provide a way to specify individual methods on a struct do not modify state, in the same way that `readonly struct` specifies no instance method modifies state.
+
+It is worth noting that `readonly method` != `pure method`. A `pure` method guarantees no state will be modified. A `readonly` method only guarantees that instance state will not be modified.
+
+All instance methods on a `readonly struct` could be considered implicitly `readonly methods`. Explicit `readonly methods` declared on non-readonly structs would behave in the same manner. For example, they would still create hidden copies if you called an instance method (on the current instance or on a field of the instance) which was itself not-readonly.
```

Comment:
Probably worth checking that this statement matches what you'd expect. It is basically saying that all instance methods on a `readonly struct` are implicitly `readonly methods`.

Reply:
I think somewhere we should add the more explicit language explanation. Essentially that marking an instance member as `readonly` causes the type of the `this` parameter to be `in T` instead of `ref T`.